src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -245,37 +244,19 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
         // Fetch the active interpreter and use the matching controller
         const api = await this.pythonApi.getApi();
         const activeInterpreter = await api.getActiveInterpreter();
-        const interpreterControllers = this.notebookControllerManager.registeredNotebookControllers().filter(
-            (controller) =>
-                // We register each of our kernels as two controllers
-                // because controllers are currently per-viewtype. Find
-                // the one for the interactive viewtype for now
-                controller.controller.notebookType === InteractiveWindowView
-        );
-
-        // First try to match on the kernelSpec name. This is supposed to uniquely
-        // identify the active interpreter.
-        const kernelSpecName = getInterpreterKernelSpecName(activeInterpreter);
-        let preferredController = interpreterControllers.find((controller) => {
-            return (
-                controller.connection.kind === 'startUsingPythonInterpreter' &&
-                controller.connection.kernelSpec.name === kernelSpecName
-            );
-        });
-
-        // We do some deduplication of old kernelspecs which may result in matches
-        // being hidden, so just in case the first search on the kernelSpecName fails to
-        // return a match, fall back to matching on the interpreter path. Note that we
-        // don't want to start with just the naive path match below, because it's possible to
-        // have multiple kernelspecs associated with the same Python interpreter path.
-        if (preferredController === undefined) {
-            preferredController = interpreterControllers.find((controller) => {
+        const preferredController = this.notebookControllerManager
+            .registeredNotebookControllers()
+            .find((controller) => {
                 return (
+                    // We register each of our kernels as two controllers
+                    // because controllers are currently per-viewtype. Find
+                    // the one for the interactive viewtype for now
+                    controller.controller.notebookType === InteractiveWindowView &&
                     controller.connection.kind === 'startUsingPythonInterpreter' &&
-                    controller.connection.interpreter?.path === activeInterpreter?.path
+                    controller.connection.interpreter?.path === activeInterpreter?.path &&
+                    controller.connection.interpreter.displayName === activeInterpreter.displayName
```

Comment:
Is envName preferable over displayName? https://github.com/microsoft/vscode-python/blob/846eee870e8b7bab38172600836faedb5fb80166/src/client/activation/activationService.ts#L343

Not sure I full understand the depth of when one of the other will be available. But using a display name for any type of checking seems dicy.

Reply:
`envName` is undefined for base environments while `displayName` doesn't seem to be, and I don't know enough about conda environments to know if you can have 2 base environments associated with the same Python executable but without an `envName`, but that seems hypothetically possible, which is why I ended up going with `displayName` here. But per your review comment below there would be a `displayName` collision for that case.